src/registrar/models/domain_application.py

```diff
@@ -502,6 +502,25 @@ def submit(self):
         # This is a side-effect of the state transition
         self._send_confirmation_email()
 
+    @transition(field="status", source=[SUBMITTED, INVESTIGATING], target=APPROVED)
+    def approve(self):
+        """Approve an application that has been submitted.
```

Comment:
This doesn't appear to have run when I changed the state of a domain application from the admin dashboard. While trying to debug this, I've realized that we need to add the new UserDomainRole to the admin, because otherwise I can't see whether I've been added or not.

Reply:
We used `protected=False` when we defined this FSMField so the admin is able to change the status field without going through the state transition methods. It seems now like this might not be what we want? There is also https://github.com/gadventures/django-fsm-admin which is an FSM-aware module for the Django admin that I think will run the transition methods when the state is changed in the admin.